src/System.Data.SqlClient/src/System/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1812,7 +1812,18 @@ internal void OnEnvChange(SqlEnvChange rec)
                     break;
 
                 case TdsEnums.ENV_PROMOTETRANSACTION:
-                    PromotedDTCToken = rec.newBinValue;
+                    byte[] dtcToken = null; 
+                    if (rec.newBinRented)
```

Comment:
This change is a bit unclear to me. Could you explain please? Why do we need to copy the newBinValue to dtcToken and then reference it to PromotedDTCToken? can't we directly copy to PromotedDTCToken?

Reply:
With this method PromotedDTCSession is never set until it has the correct contents. If we allocate and then copy into it directly there is a small windows where it is set but the value is incorrect.

Reply:
I see. Thanks for the explanation. What is the scenario which would cause the value to be incorrect?

Reply:
If the array contents neds to be kept the contents is copied in then assigned so there are only two states, null or array containing the correct data. otherwise there's a third state which is non-null with zeroed data.